File: sum_array_recursion/array_recursion.cpp
// Asilbek Asqarov
// Mohirdev Maʻlumotlar Tuzilmasi va Algoritmlar darslari
// dars - 08
// Sum of elements in array with recursion

#include <iostream>
#include <vector>

int sum_of_array(std::vector<int> &arr)
{
    if (arr.empty())
    {
        return 0;
    }
    int last_value = arr.back();
    arr.pop_back();
    return last_value + sum_of_array(arr);
}

int main()
{
    std::vector<int> arr = {5, 8, 12, 22};
    std::cout << "Sum of elements in array: " << sum_of_array(arr) << std::endl;
}

File: sum_array_recursion/array_recursion.py
#Asilbek Asqarov
#Mohirdev Maʻlumotlar Tuzilmasi va Algoritmlar darslari
#dars-08
#Sum of elements in array with recursion

def sum_of_array(nums:list):
    if len(nums) == 0:
        return 0
    elif len(nums) == 1:
        return nums[0]
    return nums.pop() + sum_of_array(nums)



def sum_array(nums:list):
    if not nums:
        return 0
    return nums.pop() + sum_array(nums)

arr = [5, 8, 12, 22] 
print(sum_of_array(arr))
arr = [5, 8, 12, 22] 
print(sum_array(arr))
